### DISCUSSION TITLE
Rows with checkbox + slot

### DISCUSSION THREAD

**Discussion by JoseGonzalz** [Telefonica/mistica-design#1647] [Proposals]
### Goal

Poder usar el slot zone con cualquier preset (navigation, checkbox, radio, switch...) en las rows/boxed rows.

### Context

Actualmente en las rows/boxed rows tenemos una opción en la "zona C - Presets" para mostrar "navigation + slot", pero cuando queremos cambiar a otro componente como checkbox, no podemos poner "slot zone" y solo podemos poner "details"

<img width="1083" alt="Captura de pantalla 2024-02-12 a las 11 09 07" src="https://github.com/Telefonica/mistica-design/assets/129079621/fbcc031a-9d87-450b-83a9-805e2e357717">

En DIPA, en la selección de linea, usamos ese slot en unas rows con un diseño concreto, pero al continuar en el flujo y seleccionar las facturas de esa linea y al tener que poner un checkbox para seleccionar, solo podemos usar la opción de detalle y no slot, con lo que visualmente se rompe la homogeneidad del precio.

<img width="806" alt="Captura de pantalla 2024-02-12 a las 11 13 20" src="https://github.com/Telefonica/mistica-design/assets/129079621/67737f1d-cb82-4931-bfe4-d60484f91aaf">

Esta necesidad también la están teniendo en MOVES con todo lo relacionado a descarga de facturas.

### Impact

Poder añadir slot zone no solo con navigation preset, si no con cualquier otro (navigation, checkbox, radio, switch...) para homogeneizar.

<img width="807" alt="Captura de pantalla 2024-02-12 a las 11 16 44" src="https://github.com/Telefonica/mistica-design/assets/129079621/f1991d7e-ce96-4caf-8ab2-27306030e39b">

### Discussion Questions

- Deberíamos poder usar slot con cualquier preset y homogeneizar las opciones que tienes cuando seleccionas navigation.

### Do you have a visual proposal?

https://www.figma.com/file/cL8KKUna805Zf9ezYjML1r/Mistica-Requests?type=design&node-id=1-465&mode=design

**Comment by yceballost** [comment]
Por qué no usas el textSecondary con el mismo estilo del detail para homogeneizar? Si estoy entendiendo bien, lo que estás pidiendo es abrir el estilo de detail para poner un estilo diferente?

**Reply by JoseGonzalz** [reply]
Necesitamos poner la información del precio de una forma más relevante que como se muestra con el detail. Es de las informaciones más importantes en esas pantallas y con el estilo del detail se queda en un segundo plano.

**Reply by AnaMontes11** [reply]
Hola @JoseTelefonica te compartimos por aquí algunas alternativas que estamos estudiando para solucionar el problema que comentas.

Todas tienen sus pros y contras, por lo que necesitamos analizarlas en profundidad:

1. Habilitar la row para que pueda mostrar el slot (extra) cuando hay un checkbox u otros controles.
2. Ampliar el uso del Detail. Es decir, permitir un Detail más abierto que puedas “customizar” a partir de unas propiedades definidas desde Mística. Podría ser un approach dado que es el Detail el que no se ajusta a tu necesidad, pero hay implicaciones técnicas que nos bloquean para elegir esta propuesta.
3. Habilitar un slot en el detail. Esto abre la posibilidad de tener un slot en el slot lo que podría crear confusión en cuanto a cuando usar uno u otro.

Para ti, ¿cual de ellas crees que puede encajar más?
<img width="582" alt="Captura de pantalla 2024-02-14 a las 12 47 13" src="https://github.com/Telefonica/mistica-design/assets/88187691/b84515b5-6944-48db-9b01-489f363177fa">

**Reply by JoseGonzalz** [reply]
Para mi la que más sentido tiene es la 2º "Habilitar slot+checkbox" porque esa opción ya existe si eliges navigation, con lo que me parece que la mejor forma de estandarizar eso es que puedas tener slot + preset (de cualquier tipo; navigation, checkbox, switch, radio, icon with action o ninguno incluso)

**Reply by AnaMontes11** [reply]
Después de comentarlo con el equipo de diseño, vemos que la solución de habilitar `slot+controls` es la que más se ajusta a su necesidad. Creamos tarea para evolucionarlo. Gracias @JoseTelefonica

**Reply by AnaMontes11** [reply]
Esta tarea será cubierta en la issue #1653. Cierro discussion.
